analysis/analysis.go

```diff
@@ -483,6 +482,37 @@ func assessMetricStatus(metric v1alpha1.Metric, result v1alpha1.MetricResult, te
 	return v1alpha1.AnalysisPhaseRunning
 }
 
+func assessMetricFailureInconclusiveOrError(metric v1alpha1.Metric, result v1alpha1.MetricResult) (v1alpha1.AnalysisPhase, []string) {
```

Comment:
If we add the message as another return value for `assessMetricStatus`, I don't think we will need this function since its functionality will be covered by `assessMetricStatus`. I think having all the assessment of the metric Status makes it easier for developers to easier to see how the analysis run is evaluated.